notify/notify.go

```diff
@@ -662,8 +668,16 @@ func NewRetryStage(i Integration, groupName string, metrics *Metrics) *RetryStag
 func (r RetryStage) Exec(ctx context.Context, l log.Logger, alerts ...*types.Alert) (context.Context, []*types.Alert, error) {
 	r.metrics.numNotifications.WithLabelValues(r.integration.Name()).Inc()
 	ctx, alerts, err := r.exec(ctx, l, alerts...)
+
+	statusCodeCategory := defaultStatusCodeCategory
```

Comment:
For integrations that don't implement `ErrorWithStatusCode`, this could be inaccurate (e.g. the receiving side might have returned a 400 status code). It's also not correct if Alertmanager couldn't even send the notification (template error for instance). And the email notifier is another case that doesn't fit here.

I wonder if we shouldn't abstract away from the HTTP situation and consider a more generic label like `reason` with a handful number of possible values: 
* `server error` (e.g. 5xx)
* `client error` (e.g. 4xx)
* `authentication error`
* `template`
* `other`

Reply:
@simonpasquier would this work?

Reply:
I think that we can start with:
* ClientError (typically templating errors, 4xx responses)
* ServerError (typically 5xx responses from the server)
* Other (e.g. receivers for which we haven't implemented ClientError/ServerError logic yet)

And we can always add more in the future where we see fit (e.g. maybe we'll want to distinguish authn failures, serialization errors, templating errors, ...)

Reply:
done